src/components/TimePeriodFilter.js

```diff
@@ -2,26 +2,45 @@ import React from 'react'
 
 const MIN_YEAR = 1960
 const MAX_YEAR = 2014
-const YEAR_WINDOW = 10
 
 class TimePeriodFilter extends React.Component {
   constructor(props) {
     super(props)
     this.handleChange = ::this.handleChange
-    this.state = { error: null }
+    this.setError = ::this.setError
+    this.state = {
+      error: null,
+      timeFrom: this.props.timeFrom,
```

Comment:
do u feel like it's necessary to store these `timefrom` and `timeto` values in the state vs. just having them as props -- it seems like we can maybe do the same checks (in the `handleChange`) while keeping the state smaller